pkgs/applications/office/beancount/default.nix

```diff
@@ -0,0 +1,38 @@
+{ stdenv, fetchhg, pkgs, pythonPackages }:
+
+pythonPackages.buildPythonApplication rec {
+  version = "2016-04-08-94915c82079e";
+  name = "beancount-${version}";
+  namePrefix = "";
+
+  src = fetchhg {
+    url = "https://bitbucket.org/blais/beancount";
+    rev = "94915c82079e";
+    sha256 = "076vxrc0s3w1i731glgw2kvfgz48fijgvr0wc6d2awrqbj4ggsbg";
+  };
+
+  buildInputs = (with pkgs; [
+    libxml2
+    libxslt
+  ]) ++ (with pythonPackages; [
+    dateutil
+    bottle
+    ply
+    lxml
+    magic
+    beautifulsoup4
+  ]);
```

Comment:
by the way, are all these Python packages really buildInputs only? Are they not needed during runtime? If they are needed during runtime, you need to add them to propagatedBuildInputs instead.

Also, I now see you have libxml2 and lxml. If the package uses lxml you'll likely not need libxml2 and libxslt because they are buildInputs for lxml, unless of course (some part of) beancounter links against those two libraries.